[solved] fatal error: ospf_vty_clippy.c: No such file or directory

on cross compile if I activate ospf I get this error, only with ospf, any other `*_vty_clippy.c` is found by CC 
the `ospf_vty_clippy.c` file is present like the other files.

```
./hosttools/lib/clippy ../python/clidef.py -o ospfd/ospf_vty_clippy.c ../ospfd/ospf_vty.c
  CC       ospfd/ospf_vty.o
../ospfd/ospf_vty.c:190:29: fatal error: ospf_vty_clippy.c: No such file or directory
compilation terminated.
Makefile:3561: recipe for target 'ospfd/ospf_vty.o' failed
```

Try doing `make clean` and `git clean -fd` and see if that solves it.

On Dec 3, 2017 6:48 PM, "Lucian CRISTIAN" <notifications@github.com> wrote:

> on cross compile if I activate ospf I get this error, only with ospf, any
> other *_vty_clippy.c is found by CC
> the ospf_vty_clippy.c file is present like the other files.
>
> ./hosttools/lib/clippy ../python/clidef.py -o ospfd/ospf_vty_clippy.c ../ospfd/ospf_vty.c
>   CC       ospfd/ospf_vty.o
> ../ospfd/ospf_vty.c:190:29: fatal error: ospf_vty_clippy.c: No such file or directory
> compilation terminated.
> Makefile:3561: recipe for target 'ospfd/ospf_vty.o' failed
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FRRouting/frr/issues/1511>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGgr-6iGa40vmAn2Hb9EC8ZHiJkCaOplks5s8zM_gaJpZM4Qz9CV>
> .
>

--- a/bgpd/bgp_rpki.c
+++ b/bgpd/bgp_rpki.c
@@ -52,7 +52,9 @@
 #include "libfrr.h"
 #include "version.h"
 
-#include "bgp_rpki_clippy.c"
+#ifndef VTYSH_EXTRACT_PL
+#include "bgpd/bgp_rpki_clippy.c"
+#endif
 
 DEFINE_MTYPE_STATIC(BGPD, BGP_RPKI_CACHE, "BGP RPKI Cache server")
 DEFINE_MTYPE_STATIC(BGPD, BGP_RPKI_CACHE_GROUP, "BGP RPKI Cache server group")

--- a/bgpd/bgp_vty.c
+++ b/bgpd/bgp_vty.c
@@ -721,7 +721,7 @@ static void bgp_clear_star_soft_out(struct vty *vty, const char *name)
 
 
 #ifndef VTYSH_EXTRACT_PL
-#include "bgp_vty_clippy.c"
+#include "bgpd/bgp_vty_clippy.c"
 #endif
 
 /* BGP global configuration.  */

--- a/doc/cli.md
+++ b/doc/cli.md
@@ -210,7 +210,7 @@ all DEFPY statements**:
 
 ...
 
-#include "filename_clippy.c"
+#include "daemon/filename_clippy.c"
 
 DEFPY(...)
 DEFPY(...)

--- a/ldpd/subdir.am
+++ b/ldpd/subdir.am
@@ -38,6 +38,7 @@ ldpd_libldp_a_SOURCES = \
 	ldpd/util.c \
 	# end
 
+ldpd/ldp_vty_cmds_clippy.c: $(CLIPPY_DEPS)
 ldpd/ldp_vty_cmds.$(OBJEXT): ldpd/ldp_vty_cmds_clippy.c
 
 noinst_HEADERS += \

--- a/lib/plist.c
+++ b/lib/plist.c
@@ -1321,7 +1321,7 @@ static int vty_clear_prefix_list(struct vty *vty, afi_t afi, const char *name,
 }
 
 #ifndef VTYSH_EXTRACT_PL
-#include "plist_clippy.c"
+#include "lib/plist_clippy.c"
 #endif
 
 DEFPY (ip_prefix_list,

--- a/ospfd/ospf_vty.c
+++ b/ospfd/ospf_vty.c
@@ -187,7 +187,7 @@ static void ospf_show_vrf_name(struct ospf *ospf, struct vty *vty,
 }
 
 #ifndef VTYSH_EXTRACT_PL
-#include "ospf_vty_clippy.c"
+#include "ospfd/ospf_vty_clippy.c"
 #endif
 
 DEFUN_NOSH (router_ospf,

--- a/sharpd/sharp_vty.c
+++ b/sharpd/sharp_vty.c
@@ -29,7 +29,9 @@
 
 #include "sharpd/sharp_zebra.h"
 #include "sharpd/sharp_vty.h"
+#ifndef VTYSH_EXTRACT_PL
 #include "sharpd/sharp_vty_clippy.c"
+#endif
 
 extern uint32_t total_routes;
 extern uint32_t installed_routes;

